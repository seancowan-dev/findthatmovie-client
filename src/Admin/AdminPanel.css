.active {
    display: block;
  }
  .inactive {
    display: none;
  }

  .admin-panel {
    margin-top: 30px;
  }