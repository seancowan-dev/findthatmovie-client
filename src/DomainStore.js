import {observable,action,toJS} from 'mobx';
import uuid from "uuid";

class DomainStore{
    // App Primary 
     @observable updateType = "";
     @observable updateFolderId = "";
     @observable updateNoteId = "";
     @observable currentNote = null;
     @observable done = "";
     @observable updated = null;
     @observable errorMsg = "";
     @observable notes = {};
     @observable folders = {};

    // Update Component
     @observable folderNotes = [];
     @observable selectedFolder = "";
     @observable selectedFolderId = "";
     @observable selectedFolderIdOld = "";
     @observable oldNoteId = null;
     @observable currentNoteId = null;
     @observable migrateChecked = false;
     @observable note = {
         id: null,
         name: null,
         folderid: null,
         modified: null,
         content: null
     };

    // Local methods to use with fetch api methods
     handleErrors(response) { // prepares error message for HTTP request errors
      if (response.ok === true) {
          return response.json();
      } else {
          throw new Error("Code " + response.status + " Message: " + response.statusText)
      }
    }
    hex(string) {
      let source = unescape(encodeURIComponent(string));
      let hex = '';
        for (var i = 0; i < source.length; i++) {
            hex += source.charCodeAt(i).toString(16)
        };
        return hex;
    }

  // Fetch API Methods

  @action async getFolders() {
    return await fetch(`http://localhost:8000/api/folders`, {
      method: 'GET',
      headers: {
        'content-type': 'application/json'
      },
    })
    .then(response => this.handleErrors(response))
    .then(resJSON => { return resJSON })
    .catch(e => alert(e));
  }

  @action async getNotes() {
    return await fetch(`http://localhost:8000/api/notes`, {
      method: 'GET',
      headers: {
        'content-type': 'application/json'
      },
    })
    .then(response => this.handleErrors(response))
    .then(resJSON => { return resJSON })
    .catch(e => alert(e));
  }

  @action async deleteNote(id) {
    return await fetch(`http://localhost:8000/api/notes/${id}`, {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json'
      },
    })
    .then(response => { 
      if (response.ok === true) {
        this.promises();
    } else {
        throw new Error("Code " + response.status + " Message: " + response.statusText)
    }
    })
    .catch(e => alert(e));

  }

  @action async addNote(event) {
    event.preventDefault();
    let workingArr = Array.from(event.target.parentNode.childNodes);
    let date = new Date(Date.now());
    let note = {
      id: uuid.v4(),
      name: "",
      modified: date.toISOString(),
      folderid: "",
      content: ""
    };
    workingArr.filter(child => {
      if(child.name === "add-note-content") {
        note.content = child.value;
      }
      if (child.name === "add-note-title") {
        note.name = child.value;
      }
      if (child.name === "select-note-folder") {
        note.folderid = child.selectedOptions[0].id;
      }
      return null
    });
    await fetch(`http://localhost:8000/api/notes/`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify(note)
    })
    .then(response => {
      if (response.ok === true) {
        this.promises()
    } else {
        throw new Error("Code " + response.status + " Message: " + response.statusText)
    }
    }).catch(e => alert(e));    
  }

  @action async addFolder(event) {
    event.preventDefault();
    let workingArr = Array.from(event.target.parentNode.childNodes);
    let folder = {
      id: "",
      name: "",
    };
    workingArr.filter(child => {
      if(child.name === "add-folder-title") {
        folder.name = child.value;
        folder.id = uuid.v4();
      }
      return null
    });

    await fetch(`http://localhost:8000/api/folders/`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify(folder)
    })
    .then(response => {
      if (response.ok === true) {
        this.promises()
    } else {
        throw new Error("Code " + response.status + " Message: " + response.statusText)
    }
    }).catch(e => alert(e));
  }

  @action async deleteFolder(id) {
    await fetch(`http://localhost:8000/api/folders/${id}`, {
      method: `DELETE`,
      headers: {
        'content-type': 'application/json'
      },

    })
    .then(response => { 
      if (response.ok === true) {
        this.promises()
    } else {
        throw new Error("Code " + response.status + " Message: " + response.statusText)
    }
    })
    .catch(e => alert(e));
  }

  @action async updateFolder(id, updates) {
    await fetch(`http://localhost:8000/api/folders/${id}`, {
      method: `PATCH`,
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify(updates),
    })
    .then(response => {
      if (response.ok === true) {
        this.promises()
    } else {
        throw new Error("Code " + response.status + " Message: " + response.statusText)
    }
    })
    .catch(e => alert(e));
  }

  @action async updateNote(id, updates) {
    await fetch(`http://localhost:8000/api/notes/${id}`, {
      method: `PATCH`,
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify(updates),
    })
    .then(response => {
        if (response.ok === true) {
          this.promises()
        } else {
          throw new Error("Code " + response.status + " Message: " + response.statusText)
        }
    })
    .catch(e => alert(e));
  }

  @action promises() {
    Promise.all([this.getFolders(), this.getNotes()]).then(res => {
      this.folders = res[0];
      this.notes = res[1];
      this.done = true;
    });
  }
}
export default new DomainStore();