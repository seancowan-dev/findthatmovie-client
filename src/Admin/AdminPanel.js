import React from 'react';
import { observer, inject } from 'mobx-react';
import { Message } from 'semantic-ui-react'; 
import BuildBaseTable from './comps/buildBaseUserTable';
import AccNav from '../comps/account-nav/AccountNav';
import uuid from 'uuid';
import './AdminPanel.css';
import AccessDenied from '../comps/access-denied/AccessDenied';

const AdminPanel = inject('userStore', 'helpers')(observer((props) => {
    let table
    if (props.userStore.adminUserList !== undefined && props.userStore.adminUserList !== null) { // Check to see if there are any admin users in the database
        table = <BuildBaseTable /> // If so build the base admin table
    
        return (<>
            <div className="container admin-panel">
                <AccNav />
                <h1>Administrator Control Panel</h1>
                <h4>Add/Manage Users</h4>
                <Message positive key={uuid.v4()} floating className={props.helpers.checkMessageVisible(props.userStore.adminPanelMessage.visible)} content={props.userStore.adminPanelMessage.message} />
                {table}
            </div>
            </>);
    }
    
    else {
        return (<>
            <div className="container admin-panel">
                <AccessDenied userType="site Administrators" />
            </div>
            </>) 
    }
}));

export default AdminPanel;