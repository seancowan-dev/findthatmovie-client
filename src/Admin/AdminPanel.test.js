import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from "mobx-react";
import DomainStore from '../DomainStore';
import AdminPanel from './AdminPanel';

const store = {
    dataStore: DomainStore.dataStore,
    searchStore: DomainStore.searchStore,
    helpers: DomainStore.helpers,
    userStore: DomainStore.userStore,
    validators: DomainStore.validators
  }

describe('| AdminPanel Test Object |', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Provider {...store}><AdminPanel debug /></Provider>);
    expect(component).toMatchSnapshot();
  });
  it('should render correctly in "production" mode', () => {
    const component = shallow(<Provider {...store}><AdminPanel production /></Provider>);
    expect(component).toMatchSnapshot();
  });
});