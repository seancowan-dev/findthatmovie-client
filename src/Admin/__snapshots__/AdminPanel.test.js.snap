// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`| AdminPanel Test Object | should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`| AdminPanel Test Object | should render correctly in "production" mode 1`] = `ShallowWrapper {}`;
